From 97d6df1ca09102ca95ada1ae3ef9284881b2130c Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Mon, 27 Apr 2026 14:52:49 +0000
Subject: [PATCH] [client,x11] fix xf_rail_window_common cleanup

Backport of commit b4f0f0a18fe53aa8d47d062f91471f4e9c5e0d51.

Adjusted hunk offsets for 3.10.3.

Made-with: Cursor
---
 client/X11/xf_rail.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/client/X11/xf_rail.c b/client/X11/xf_rail.c
index 5c42f7a..afcfc67 100644
--- a/client/X11/xf_rail.c
+++ b/client/X11/xf_rail.c
@@ -307,11 +307,10 @@ static void window_state_log_style_int(wLog* log, const WINDOW_STATE_ORDER* wind
 static BOOL xf_rail_window_common(rdpContext* context, const WINDOW_ORDER_INFO* orderInfo,
                                   const WINDOW_STATE_ORDER* windowState)
 {
-	xfAppWindow* appWindow = NULL;
 	xfContext* xfc = (xfContext*)context;
 	UINT32 fieldFlags = orderInfo->fieldFlags;
 	BOOL position_or_size_updated = FALSE;
-	appWindow = xf_rail_get_window(xfc, orderInfo->windowId);
+	xfAppWindow* appWindow = xf_rail_get_window(xfc, orderInfo->windowId);
 
 	if (fieldFlags & WINDOW_ORDER_STATE_NEW)
 	{
@@ -362,10 +361,7 @@ static BOOL xf_rail_window_common(rdpContext* context, const WINDOW_ORDER_INFO*
 		}
 
 		if (!appWindow->title)
-		{
-			free(appWindow);
 			return FALSE;
-		}
 
 		xf_AppWindowInit(xfc, appWindow);
 	}
-- 
2.53.0

